File: Backend/src/main/java/model/Booking.java
package model;

import jakarta.persistence.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
public class Booking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "ticket_id")
    private Ticket ticket;

  //  @OneToOne
   // @JoinColumn(name = "customer_id")
  //  private Customer customer;

 //   @ManyToOne
  //  @JoinColumn(name = "vendor_id")
    //private Vendor vendor;

    //private String bookingType;
   // private LocalDateTime dateGiven;


}

File: ticketing-system-backend/models/booking.js
const mongoose=require('mongoose');


const bookingSchema=new mongoose.Schema({
    vendor:{
        type:mongoose.Schema.Types.ObjectId,
        ref:'Vendor'

    },

    customer:{
        type:mongoose.Schema.Types.ObjectId,
        ref:'Customer'

    },

    ticket:{
        type:mongoose.Schema.Types.ObjectId,
        ref:'Ticket'
    },

    bookingType:String,
    dateGiven:{
        type:Date,
        default:Date.now
    }


});
module.exports=mongoose.model('Booking',bookingSchema);